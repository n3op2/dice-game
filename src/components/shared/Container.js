import React from 'react';

const defaultStyle = {
    display: 'flex',
    marginTop: 20,
    marginBottom: 20,
    flexDirection: 'column',
    alignItems: 'center',
};

const Container = (props) => (
    <div style={{ ...defaultStyle, ...props.style }}>{props.children}</div>
);

export default Container;
