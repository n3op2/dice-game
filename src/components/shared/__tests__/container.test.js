import React from 'react';
import { shallow } from 'enzyme';
import Container from '../Container';

const Component = (
    <Container>
        <ul>
            <li></li>
        </ul>
        <div className="a">some text</div>
    </Container>
);

describe('Container Component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(Component);
    });

    test('overrites default style with supplied props', () => {
        wrapper = shallow(
            <Container style={{ display: 'none', flexDirection: 'row' }} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders a component with all child nodes', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
