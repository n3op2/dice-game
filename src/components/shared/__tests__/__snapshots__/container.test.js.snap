// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component overrites default style with supplied props 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "none",
      "flexDirection": "row",
      "marginBottom": 20,
      "marginTop": 20,
    }
  }
/>
`;

exports[`Container Component renders a component with all child nodes 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": 20,
      "marginTop": 20,
    }
  }
>
  <ul>
    <li />
  </ul>
  <div
    className="a"
  >
    some text
  </div>
</div>
`;
